src/main/java/de/tum/cit/aet/artemis/programming/web/ProgrammingExerciseResource.java

```diff
@@ -492,6 +498,21 @@ public ResponseEntity<ProgrammingExercise> getProgrammingExercise(@PathVariable
         return ResponseEntity.ok().body(programmingExercise);
     }
 
+    /**
+     * GET /programming-exercises/:exerciseId/build-config : get the build config of "exerciseId" programmingExercise.
+     *
+     * @param exerciseId the id of the programmingExercise to retrieve the config for
+     * @return the ResponseEntity with status 200 (OK) and with body the programmingExerciseBuildConfig, or with status 404 (Not Found)
+     */
+    @GetMapping("programming-exercises/{exerciseId}/build-config")
+    @EnforceAtLeastStudentInExercise
+    public ResponseEntity<ProgrammingExerciseBuildConfig> getBuildConfig(@PathVariable long exerciseId) {
```

Comment:
We want to use DTOs. This way you also don't have to change anything about the JSON config annotations.

Reply:
There is not DTO for it and the Exercise is not using a DTO itself (as far as I have seen it in the respective services). You can take a look at how exercises are being loaded to catch my drift :)